### PR TITLE
fix(toggl-plan): Extract Description & Project from the Plan event

### LINKED ISSUE #1793: When starting time entry from Plan task, description is not filled
* **OS version**: macOS
* **Browser version**: Chrome 83.0.4103.116
* **Extension version**: 1.64.1 

#### Relevant integration (if any):

Toggl Plan 

### 🐛 Describe the bug
When starting a TE from inside Plan, the description field isn't filled out. 

image

### Expected behaviour

The description should be task name. 

### Steps to reproduce
1. Open task in Plan
2. Click TB in task 

### Other details or context

[Slack](https://toggl.slack.com/archives/C9LMUPD9T/p1595407417143500)

[Case reference 1
](https://app.intercom.com/a/apps/ayixs927/inbox/inbox/534741/conversations/106481300033931)
[Case reference 2](https://app.intercom.com/a/apps/ayixs927/inbox/inbox/534741/conversations/106481300034052)

### PR DESCRIPTION
**Pull request by kornelijepetak**
## :star2: What does this PR do?
Fills in the description (from the Event name) and the project (from the Plan name, if available) fields in Toggl Plan TB integration.

## :bug: Recommendations for testing

* Open any Toggl Plan event
* Start a TB timer and check that the description matches the event description
* Make yourself a project that would match the plan name.
* Start a TB timer again and check that the project was automatically selected as well.
* If no project exists or a plan is not selected, the project field should be empty (no selection).

## :memo: Links to relevant issues or information

Closes #1793

### CODE REVIEW

**Comment by kornelijepetak** [comment]
Note that I have _replaced_ the previous integration, as old teamweek is no longer reachable.

Also, if you think that Plan does not correspond meaningfully with Time entry's project field, we can remove that.

File: src/scripts/content/toggl-plan.js [hunk 1]
```diff
@@ -1,24 +1,32 @@
...
-    className: 'teamweek-new',
-    buttonType: 'minimal',
-    description: getDescriptionText,
-    projectName: getProjectText
-  });
+      if (!planElement) { return ''; }
 
-  container.parentNode.insertBefore(link, container);
-});
+      return planElement ? planElement.textContent.trim() : null;
```
[the remark this answers is not in the document]
Comment (kornelijepetak) [reply src/scripts/content/toggl-plan.js]: I derped cause I was redoing this and I accidentally forgot to remove that if condition altogether. Nice catch.

**Comment by kornelijepetak** [comment]
@rylek90 btw. I added `left-margin` instead, because that's what other buttons to the top do, just for consistency. The same amount as well.

File: src/scripts/content/toggl-plan.js [hunk 2]
```diff
@@ -1,24 +1,35 @@
...
-});
+    const link = togglbutton.createTimerLink({
+      className: 'toggl-plan',
+      buttonType: 'minimal',
+      description: getDescriptionText,
+      projectName: getProjectText
+    });
+
+    link.style.marginTop = '-1px';
+    link.style.marginLeft = '16px';
```
Comment (rylek90) [inline src/scripts/content/toggl-plan.js:31]: Totally not RFC: we usually define in `src/styles/style.css`

File: src/scripts/content/toggl-plan.js [hunk 3]
```diff
@@ -1,24 +1,35 @@
...
-  });
+      return planElement
+        ? planElement.textContent.trim()
+        : '';
+    };
 
-  container.parentNode.insertBefore(link, container);
-});
+    const link = togglbutton.createTimerLink({
+      className: 'toggl-plan',
```
Comment (rylek90) [inline src/scripts/content/toggl-plan.js:24]: This class doesn't exist in `src/styles/style.css` anymore 🤔

**Pull request merged**